src/cc_catalog_airflow/dags/provider_api_scripts/noun_project.py

```diff
@@ -0,0 +1,269 @@
+    auth=AUTH,
+    endpoint=ENDPOINT,
+    retries=3,
+    total_pages=2366  # Change this when total pages for collections changes
+):
+    collections_list = []
+    page = 1
+    while (page <= total_pages):
+        query_params = _build_query_param(page=page)
+        page += 1
+        json_response_inpydict_form = delayed_requester.get_response_json(
+            endpoint=(endpoint + "collections"),
+            retries=retries,
+            query_params=query_params,
+        )
+
+        _collections_list = _extract_collections_list_from_json(
+            json_response_inpydict_form
+        )
+        if _collections_list is None:
+            break
+        for collection in _collections_list:
+            collections_list.append(collection)
+
+    return collections_list
+
+
+def _build_query_param(page=1):
+    query_params = {}
+    query_params.update({"page": page})
+
+    return query_params
+
+
+def _extract_collections_list_from_json(
+        json_response_inpydict_form
+):
+    if (
+        json_response_inpydict_form is None
+        or json_response_inpydict_form.get("collections") is None
+        or len(json_response_inpydict_form.get("collections")) == 0
+    ):
+        collections_list = None
+    else:
+        collections_list = []
+        _collections_list = json_response_inpydict_form.get("collections")
+        for collection in _collections_list:
+            collections_list.append(collection.get("slug"))
+
+    return collections_list
+
+
+def _get_icons_list_from_collection(
+    collection=None,
+    retries=3,
+    endpoint=ENDPOINT,
+    auth=AUTH
+):
+    icon_count = get_icon_count(collection=collection)
+    total_pages = _find_no_of_pages_for_specific_collection(icon_count)
+    icons_list = []
+    page = 1
+    while (page <= total_pages):
+        query_params = _build_query_param(page=page)
+        page += 1
+        json_response_inpydict_form = delayed_requester.get_response_json(
+            # Request is of the form:
+            # "http://api.thenounproject.com/collection/{coll_name}/
+            # icons?page={page_no}"
+            endpoint=(endpoint + "collection/" + collection + "icons"),
+            retries=retries,
+            query_params=query_params,
+        )
+
+        _icons_list = _extract_icons_list_from_json(
+            json_response_inpydict_form
+        )
+        if _icons_list is None:
+            break
+        for icon in _icons_list:
+            icon.update({"collection": collection})
+            icons_list.append(icon)
+
+    if len(icons_list) == 0:
+        logger.warning("No more tries remaining. Returning Nonetypes.")
+        return None
+    else:
+        return icons_list
+
+
+def _find_no_of_pages_for_specific_collection(icon_count=None):
+    pages = 1
+    if icon_count is None:
+        return pages
+
+    n = icon_count / 50
+    if n != 0:
+        pages = int(icon_count//50 + 1)
+    else:
+        pages = int(icon_count/50)
```

Comment:
If you're using the default fallback for pages as 1, is there any specific reason to set this to return 0, in case of n=0?